Add Transit Gateway setup steps (#882)
astronomer/docs: astro/connect-external-services.md

If your target VPC resolves DNS hostnames using [private hosted zones](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/hosted-zones-private.html), then you must associate your Route53 private hosted zone with the Astronomer VPC using instructions provided in [AWS Documentation](https://aws.amazon.com/premiumsupport/knowledge-center/route53-private-hosted-zone/). You can retrieve the ID of the Astronomer VPC by contacting [Astronomer support](https://support.astronomer.io).

> ## Set up Transit Gateway (_AWS only_)

Reviewer: ```suggestion
## Set up an AWS Transit Gateway
```

Reviewer 2: The other sections are called "Public Endpoints", "VPC Peering", so calling this "Set up an AWS Transit Gateway" seems off.

> Use AWS Transit Gateway to connect your Astro clusters to your on-premises networks.

Reviewer 2: Use AWS Transit Gateways to connect your Astro Clusters to your VPCs and on-premises networks

[i.e. it is possible - and much more likely - that our customers use Transit Gateways as a replacement for VPC peering - i.e. without any on-premises connectivity]

### Prerequisites

- An Astro cluster
- A transit gateway in the same region as your Astro cluster
- Permission to share resources using AWS Resource Access Manager (RAM)

:::info

If your transit gateway is in a different region than your Astro cluster, contact [Astronomer support](https://support.astronomer.io). Astronomer can create a new transit gateway and set up cross-region peering with your existing transit gateway. Note that this setup can incur AWS charges for a new transit gateway hosted in the Astro account, as well as inter-region transfer costs.

:::

### Setup

> To implement Transit Gateway for a cluster:

Reviewer: ```suggestion
To implement Transit Gateway for a cluster:
```

Remove this stem sentence as it doesn't add any value.

> 1. In the Cloud UI, open the **Clusters** tab and copy your cluster's **Account ID**.

Reviewer: ```suggestion
1. In the Cloud UI, click the **Clusters** tab and copy the **Account ID** for the cluster.
```

> 2. Follow the [AWS documentation](https://docs.aws.amazon.com/ram/latest/userguide/working-with-sharing-create.html) to use AWS RAM to share a transit gateway with your cluster's AWS account. The account ID that you share with is the same as the ID that you copied in Step 1.

Reviewer: ```suggestion
2. Create a resource share in AWS RAM. See [Creating a resource share in AWS RAM](https://docs.aws.amazon.com/ram/latest/userguide/working-with-sharing-create.html)  The account ID is the value that you copied in Step 1.
```

Author: I'm adding back "that you share with" because otherwise it seems like we mention account ID out of nowhere (in relation to resource sharing)

> 2. Follow the [AWS documentation](https://docs.aws.amazon.com/ram/latest/userguide/working-with-sharing-create.html) to use AWS RAM to share a transit gateway with your cluster's AWS account. The account ID that you share with is the same as the ID that you copied in Step 1.

Reviewer 2: Capitalize Transit Gateway for consistency?

Author: AWS Transit Gateway is the product name, but a "transit gateway" is an object that AWS doesn't capitalize in their own documentation: https://docs.aws.amazon.com/vpc/latest/tgw/tgw-peering.html#tgw-peering-add-route

We've been trying to be more cautious with our capitalization lately 😄

> 3. Contact [Astronomer support](https://support.astronomer.io) and provide the destination CIDR block of the on-premises network that should be routed through the transit gateway. From here, Astronomer approves the resource sharing request and creates a transit gateway peering attachment request to your on-premises network.

Author: I added "to your on-premises network" as an assumption, is this correct? Where/ with what account do they accept the request from?

Reviewer: Has _CIDR_ been defined anywhere?

Reviewer 2: Same comment as earlier - we should't call it on-premise as they could use a Transit Gateway purely for cloud to cloud connectivity.  

CIDR is one of those words (like subnet or VPC) that everybody knows but probably nobody can define! I think it may be ok to use as is because we use it in other places in our docs without defining it either.  The official definition of a CIDR is here: https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing

> 4. Follow the [AWS documentation](https://docs.aws.amazon.com/vpc/latest/tgw/tgw-peering.html#tgw-peering-accept-reject) to accept the Transit Gateway peering attachment request from your on-premises network.

Reviewer: ```suggestion
4. Accept the Transit Gateway peering attachment request from your on-premises network. See [Accept or reject a peering attachment request](https://docs.aws.amazon.com/vpc/latest/tgw/tgw-peering.html#tgw-peering-accept-reject).
```

> 5. Follow the [AWS documentation](https://docs.aws.amazon.com/vpc/latest/tgw/tgw-peering.html#tgw-peering-add-route) to create a static route from your CIDR block to the transit gateway.

Reviewer: ```suggestion
5. Create a static route from your CIDR block to the transit gateway. See [Add a route to the transit gateway route table](https://docs.aws.amazon.com/vpc/latest/tgw/tgw-peering.html#tgw-peering-add-route) to .
```

> 6. Contact [Astronomer support](https://support.astronomer.io) to confirm that you have created the static route. From here, Astronomer updates the routing table of the Astro cluster's VPC to send traffic from your CIDR block through the transit gateway.

Reviewer: Has VPC been defined anywhere?
